swarm/src/handler/one_shot.rs

```diff
@@ -213,7 +213,8 @@ where
             }
             ConnectionEvent::DialUpgradeError(DialUpgradeError { error, .. }) => {
                 if self.pending_error.is_none() {
-                    self.pending_error = Some(error);
```

Comment:
As far as I can tell, this was the only occurrence of setting `self.pending_error`. In other words, I don't think `pending_error` is used anywhere after this change. Is that correct?

Reply:
Almost; there is one more place
```rust
if let Some(err) = self.pending_error.take() {
    return Poll::Ready(ConnectionHandlerEvent::Close(err));
}
```      
The resolution of this usage is related to #3591.

Reply:
But if it is never set, then this is dead code so we can delete it straight-away!

Sorry for missing that @mxinden.